google/cloud/datastore/transaction.py

```diff
@@ -149,15 +151,23 @@ class Transaction(Batch):
     :param read_time: (Optional) Time at which the transaction reads entities.
                       Only allowed when ``read_only=True``. This feature is in private preview.
 
+    :type begin_later: bool
+    :param begin_later: (Optional) If True, the transaction will be started
+                        lazily (i.e. when the first RPC is made). If False,
+                        the transaction will be started as soon as the context manager
+                        is entered. `self.begin()` can also be called manually to begin
+                        the transaction at any time. Default is True.
+
     :raises: :class:`ValueError` if read_time is specified when
              ``read_only=False``.
     """
 
     _status = None
 
-    def __init__(self, client, read_only=False, read_time=None):
+    def __init__(self, client, read_only=False, read_time=None, begin_later=True):
```

Comment:
Just wanted to confirm that we did want this enabled by default?

It shouldn't be a breaking change in most cases, but I can think of a few edge cases where behaviors might end up different after this change

Reply:
After discussing with Baha, defaulting to False as the other languages are doing